---
title: "DGE_GLM"
author: "Melissa Naugle"
date: "12/21/2020"
output: html_document
---

#Setup
Load packages and install BiocManager if needed

```{r setup}
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("edgeR")
#install.packages("statmod")
library("edgeR")
library("statmod")
library("data.table")

setwd("~/Desktop/GitHub/RNAseq_DGE")
```

Import Data

```{r import_data}
#Import gene count data and view the first few rows
#had to add 'gene' to first header row on original txt file to prevent weird error
countsTable <- read.table(file="RSEM.isoform.counts.matrix.txt", row.names="gene", sep="\t", header=TRUE)
head(countsTable)
```

Import grouping factors


```{r groupingfactors}
#import grouping factor and create samples group matrix
reps <- read.table("replicates.txt", header = T)

```

Setup grouping factors and create List 

CoconutPoint.Control_28 CoconutPoint.Heat_35    CoconutPoint.Control_28
 [4] CoconutPoint.Heat_35    CoconutPoint.Control_28 CoconutPoint.Heat_35   
 [7] CoconutPoint.Control_28 CoconutPoint.Heat_35    Fagaalu.Control_28     
[10] Fagaalu.Heat_35         Fagaalu.Control_28      Fagaalu.Control_28     
[13] Fagaalu.Heat_35         Fagaalu.Control_28      Fagaalu.Heat_35        
[16] Fagatele.Heat_35        Fagatele.Control_28     Fagatele.Heat_35       
[19] Fagatele.Control_28     Fagatele.Heat_35        Fagatele.Control_28    
[22] Fagatele.Heat_35

```{r create list}
group <- factor(paste(reps$site,reps$treatment,sep="."))
cbind(reps,Group=group)

#create 'DGE list' object
#don't include group yet? #but r blog says to include so we include
y <- DGEList(counts=countsTable, group = group)
#MAKE SURE SAME ORDER ON REPS SHEET AS IN COUNTS MATRIX
colnames(y) <- reps$sample

names(countsTable)

```


Filtering and normalization

Jacoby's alternative filtering:

'Here, a gene is only retained if it is expressed at a count-per-million (CPM) above 0.5 in at least two samples
keep <- rowSums(cpm(y) > 0.5) >= 2
summary(keep)'

TMM normalization is performed to eliminate composition biases between libraries.

```{r norm_and_write_list}

#filter out lowly expressed genes
keep <- filterByExpr(y)

y <- y[keep,,keep.lib.sizes=FALSE]

#view summary of normalized counts
summary(keep)

#normalize library sizes (use trimmed mean m values to eliminate composition biases)
y <- calcNormFactors(y)

#not included in Jacoby's code 
#write normalized
normList <- cpm(y, normalized.lib.sizes = T)
nrow(normList)
#write.table(normList, file = "data/glmQLF_normalizedCounts.csv", sep=",", row.names=T)

```

MD plot to verify 

The performance of the TMM normalization procedure can be examined using meandifference (MD) plots. This visualizes the library size-adjusted log-fold change between two libraries (the difference) against the average log-expression across those libraries (the mean). The following MD plot is generated by comparing sample 1 against an artificial library constructed from the average of all other samples.

Ideally, the bulk of genes should be centred at a log-fold change of zero. This indicates that any composition bias between libraries has been successfully removed. This quality check should be repeated by constructing a MD plot for each sample.

```{r plots_to_verify}
#Verify TMM normalization using a MD plot
jpeg("Figures/glmQLF_plotNormalizedMD.jpg")
plotMD(cpm(y, log=TRUE), column=1) 
abline(h=0, col="red", lty=2, lwd=2) 
dev.off()
```

Make MDS plot


default settings for MDS plot: 
points <- c(15,15,16,16,17,17)
colors <- rep(c("blue", "red"), 3)

```{r MDSplot}
#Use a MDS plot to visualizes the differences between samples
#shape is site and color is treatment
colors <- c("red","red","yellow","yellow","green","green")
points <- rep(c(17, 16), 3)
#Write plot with legend to file
jpeg("Figures/glmQLF_plotMDS.jpg")
plotMDS(y, col=colors[group], pch=points[group], cex = 1.75, xlab = "PC1", ylab = "PC2") 
#legend("topleft", legend=levels(group), pch=points, col=colors, ncol=2, cex = 0.75)
dev.off()
```

#Design Matrix


```{r design_matrix}
#create design object with levels of treatments
design <- model.matrix(~0 + group)
colnames(design) <- levels(group)
design #looks good!
```

#Estimate Dispersion 

The NB dispersion is estimated using the estimateDisp function. This returns the DGEList object with additional entries for the estimated NB dispersions for all gene. These estimates can be visualized with plotBCV, which shows the root-estimate, i.e., the biological coefficient of variation for each gene.

estimateDisp has no effect on the downstream analysis, but is nevertheless very useful as it identifies genes that are outliers from the mean-NB dispersion trend.

```{r dispersion}
#recommended to estimate dispersion for GLM
y <- estimateDisp(y,design, robust = T)

y$common.dispersion #0.32755 is that good??
#visualize with BCV plot
jpeg("Figures/glmQLF_plotBCV.jpg")
plotBCV(y)
dev.off()
```

QL Dispersion 

Note that only the trended dispersion is used under the quasi-likelihood (QL) pipeline. The tagwise and common estimates are shown here but will not be used further. For the QL dispersions, estimation can be performed using the glmQLFit function. This returns a DGEGLM object containing the estimated values of the GLM coefficients for each gene, as well as the fitted mean-QL dispersion trend, the squeezed QL estimates and the prior degrees of freedom (df). These can be visualized with the plotQLDisp function.

```{r ql_dispersion}

fit <- glmQLFit(y, design, robust = T) #Setting robust=TRUE in glmQLFit is strongly recommended
head(fit$coefficients)
#plot to QL dispersions and write file
jpeg("Figures/glmQLF_plotQLDisp.jpg")
plotQLDisp(fit)
dev.off()

```












#Make Contrasts and run Differential Expression all heat control

```{r make_contrasts}
con <- makeContrasts(
  Coco_HvE = CoconutPoint.Heat_35 - CoconutPoint.Control_28,
  Alu_HvE = Fagaalu.Heat_35 - Fagaalu.Control_28,
  Tele_HvE = Fagatele.Heat_35 - Fagatele.Control_28, levels = design)
#The QL F-test is then applied to identify genes that are DE among the three groups. This combines the three pairwise comparisons into a single F-statistic and p-value. The top set of significant genes can be displayed with topTags.
anov <- glmQLFTest(fit, contrast=con)
topTags(anov)
# anov should show DGE bw control and heat common to all 3 sites. Meaning these are genes that are common to the heat stress response, regardless of pollution level 
#If all three contrasts are present in the contrast matrix, then only the log-fold changes of the first two contrasts are shown in the output of topTags.
is.de <- decideTestsDGE(anov, p.value=0.05)
summary(is.de)
tagsTblANOVA.one.allheatvcontrol <- topTags(anov, n = nrow(anov$table))$table
nrow(tagsTblANOVA.one.allheatvcontrol) #15007 = all



tagsTblANOVA.one.allheatvcontrol.filt <- tagsTblANOVA.one.allheatvcontrol[tagsTblANOVA.one.allheatvcontrol$FDR <= 0.05,]
nrow(tagsTblANOVA.one.allheatvcontrol.filt)

#write.table(tagsTblANOVA.one.allheatvcontrol.filt, file="data/DEcontigs_edger_glm/tagsTblANOVA.one.allheatvcontrol_FDR0.05.csv", sep=",", row.names=TRUE)


#make list for heatmap of log counts per million
logcpm <- cpm(y, log=TRUE, prior.count = T)

head(logcpm)
nrow(logcpm)
#write.table(logcpm, file = "data/glmlogcpm.csv", sep=",", row.names=T)

#write file for heatmap 
logcpm_FDR0.5_allheatvcontrol <- logcpm[c(row.names(tagsTblANOVA.one.allheatvcontrol.filt)), ]
nrow(tagsTblANOVA.one.allheatvcontrol.filt)
nrow(logcpm_FDR0.5_allheatvcontrol)
#7021 genes

#write.table(logcpm_FDR0.5_allheatvcontrol, file = "data/glmlogcpm_FDR0.5_allheatvcontrol.csv", sep=",", row.names=T)


```

#Pairwise DE

DE analysis between COCOPT control and COCOPT heat

About 2500-2900 DGE up and down 

```{r coco.heat_coco.control}
con <- makeContrasts(CoconutPoint.Heat_35 - CoconutPoint.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)
#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).
topTags(qlf)
#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)
#plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])
tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)
is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)
#plotSmear(tr, de.tags=rownames(tr)[is.de!=0])


tagsTblANOVA.one.cocohve <- topTags(qlf, n = nrow(anov$table))$table
nrow(tagsTblANOVA.one.cocohve)
tagsTblANOVA.one.cocohve.filt <- tagsTblANOVA.one.cocohve[tagsTblANOVA.one.cocohve$FDR <= 0.05,]
nrow(tagsTblANOVA.one.cocohve.filt)


#write.table(tagsTblANOVA.one.cocohve.filt, file="data/DEcontigs_edger_glm/tagsTblANOVA.one.cocoheatvcontrol_FDR0.05.csv", sep=",", row.names=TRUE)

#write file for heatmap 
logcpm_FDR0.5_COCOheatvcontrol <- logcpm[c(row.names(tagsTblANOVA.one.cocohve.filt)), ]
nrow(tagsTblANOVA.one.cocohve.filt)
nrow(logcpm_FDR0.5_COCOheatvcontrol)
#5579

#write.table(logcpm_FDR0.5_COCOheatvcontrol, file = "data/glmlogcpm_FDR0.5_COCOheatvcontrol.csv", sep=",", row.names=T)


```



DE analysis between ALU control and ALU heat

About 2100 - 2500 DGE up/down 

```{r alu.heat_alu.control}
con <- makeContrasts(Fagaalu.Heat_35 - Fagaalu.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)
topTags(qlf)
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)
#plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])
tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)
is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)
#plotSmear(tr, de.tags=rownames(tr)[is.de!=0])

tagsTblANOVA.one.aluhve <- topTags(qlf, n = nrow(anov$table))$table
tagsTblANOVA.one.aluhve.filt <- tagsTblANOVA.one.aluhve[tagsTblANOVA.one.aluhve$FDR <= 0.05,]
write.table(tagsTblANOVA.one.aluhve.filt, file="data/DEcontigs_edger_glm/tagsTblANOVA.one.aluheatvcontrolFDR0.05.csv", sep=",", row.names=TRUE)
nrow(tagsTblANOVA.one.aluhve.filt)

#write file for heatmap 
logcpm_FDR0.5_ALUheatvcontrol <- logcpm[c(row.names(tagsTblANOVA.one.aluhve.filt)), ]
nrow(tagsTblANOVA.one.aluhve.filt)
nrow(logcpm_FDR0.5_ALUheatvcontrol)
#4479

#write.table(logcpm_FDR0.5_ALUheatvcontrol, file = "data/glmlogcpm_FDR0.5_ALUheatvcontrol.csv", sep=",", row.names=T)

```




DE analysis between TELE control and TELE heat

About 1500 - 1900 DGE up/down 

```{r tele.heat_tele.control}
con <- makeContrasts(Fagatele.Heat_35 - Fagatele.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)
topTags(qlf)
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)
#plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])
tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)
is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)
#plotSmear(tr, de.tags=rownames(tr)[is.de!=0])



tagsTblANOVA.one.telehve <- topTags(qlf, n = nrow(anov$table))$table
nrow(tagsTblANOVA.one.telehve)
tagsTblANOVA.one.telehve.filt <- tagsTblANOVA.one.telehve[tagsTblANOVA.one.telehve$FDR <= 0.05,]
nrow(tagsTblANOVA.one.telehve.filt)

#write.table(tagsTblANOVA.one.telehve.filt, file="data/DEcontigs_edger_glm/tagsTblANOVA.one.teleheatvcontrolFDR0.05.csv", sep=",", row.names=TRUE)

#write file for heatmap 
logcpm_FDR0.5_TELEheatvcontrol <- logcpm[c(row.names(tagsTblANOVA.one.telehve.filt)), ]
nrow(tagsTblANOVA.one.telehve.filt)
nrow(logcpm_FDR0.5_TELEheatvcontrol)
#3703

#write.table(logcpm_FDR0.5_TELEheatvcontrol, file = "data/glmlogcpm_FDR0.5_TELEheatvcontrol.csv", sep=",", row.names=T)

```























##Ignore these comparisons 




































DE analysis between COCO control and TELE control

0 DGE 

```{r coco.control_tele.control}
con <- makeContrasts(CoconutPoint.Control_28 - Fagatele.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```

DE analysis between COCO control and ALU control

0 DGE 

```{r coco.control_alu.control}
con <- makeContrasts(CoconutPoint.Control_28 - Fagaalu.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```
DE analysis between TELE control and ALU control

0 DGE 

```{r tele.control_alu.control}
con <- makeContrasts(Fagatele.Control_28 - Fagaalu.Control_28, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```

DE analysis between TELE heat and ALU heat

1 down, 2 up  

```{r tele.heat_alu.heat}
con <- makeContrasts(Fagatele.Heat_35 - Fagaalu.Heat_35, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```

DE analysis between TELE heat and COCO heat

22 down, 40 up  

```{r tele.heat_coco.heat}
con <- makeContrasts(Fagatele.Heat_35 - CoconutPoint.Heat_35, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```

DE analysis between ALU heat and COCO heat

0 DGE

```{r alu.heat_coco.heat}
con <- makeContrasts(Fagaalu.Heat_35 - CoconutPoint.Heat_35, levels=design)
qlf <- glmQLFTest(fit, contrast=con)

#The top set of most significant genes can be examined with topTags. Multiplicity correction is performed by applying the Benjamini-Hochberg method on the p-values, to control the false discovery rate (FDR).

topTags(qlf)


#The total number of DE genes in each direction at a FDR of 5% can be examined with decideTestsDGE. 
is.de <- decideTestsDGE(qlf, p.value=0.05)
summary(is.de)

plotSmear(qlf, de.tags=rownames(qlf)[is.de!=0])

#is this diff than glmQLFtest

tr <- glmTreat(fit, contrast=con, lfc=log2(1.2))
topTags(tr)

is.de <- decideTestsDGE(tr, p.value=0.05)
summary(is.de)

plotSmear(tr, de.tags=rownames(tr)[is.de!=0])
```
##My Pairwise attempts to get DE contigs 

```{r contrasts}


```




##My own confusing attempts to run GLM 
Started with 'experiments with all combinations of multiple factors'
-made all logical contrasts 


DONT RUN

```{r make_all_contrasts_exp_with_all_combos_mult_factors}
#design all relevant contrasts 
#first for each site heat vs control
#then each site contrast at control
#then each site contrast at heat 

my.contrasts <- makeContrasts(
  Coco_HvE = CoconutPoint.Heat_35 - CoconutPoint.Control_28,
  Alu_HvE = Fagaalu.Heat_35 - Fagaalu.Control_28,
  Tele_HvE = Fagatele.Heat_35 - Fagatele.Control_28,
  CocovAlu_E = CoconutPoint.Control_28 - Fagaalu.Control_28,
  CocovTele_E = CoconutPoint.Control_28 - Fagatele.Control_28,
  AluvTele_E = Fagaalu.Control_28 - Fagatele.Control_28,
  CocovAlu_H = CoconutPoint.Heat_35 - Fagaalu.Heat_35,
  CocovTele_H = CoconutPoint.Heat_35 - Fagatele.Heat_35,
  AluvTele_H = Fagaalu.Heat_35 - Fagatele.Heat_35, 
  levels=design)

colnames(fit)



#glm for coco heat v control 
test.anov.one <- glmQLFTest(fit, contrast= my.contrasts[,"Coco_HvE"])
#Could continue from here

```

Used 'interaction at any time' 
Try using the number method for contrasts
compares each group to 28C 
but also compares to coconut point for some reason??


```{r contrasts_interactions_any_time}

reps$treatment <- relevel(reps$treatment, ref="Control_28")
#form design matrix
design <- model.matrix(~ site, data=reps)
fit_glm <- glmQLFit(y, design)
colnames(fit_glm)
qlf <- glmQLFTest(fit, coef=2:3)
#not quite right- comparing to coco as control ?
topTags(qlf)
FDR <- p.adjust(qlf$table$PValue, method="BH")
sum(FDR < 0.05)
```






'Comparisons both between and within subjects'

```{r comparisons_both_btwn_among_subjects}
#define experimental factors
Site <- factor(reps$Site, levels=c("CoconutPoint","Fagaalu", "Fagatele"))
Treatment <- factor(reps$treatment, levels=c("Control_28","Heat_35"))
genotype <- factor(reps$genotype)
#adjust for baseline genotype diffs
design <- model.matrix(~genotype)
#stopped here because stopped looking promising :(
```







Based on R bloggers guide
Not sure if this is allowed but tried to do all heats - all controls

```{r contrasts_heatvcontrol}
#Design a contrast to test overall effect of the first factor 
#i choose treatment first
#so all heats - controls
con.heatvscontrol <- makeContrasts(heatvscontrol = (CoconutPoint.Heat_35 + Fagaalu.Heat_35 + Fagatele.Heat_35)/3 - (CoconutPoint.Control_28 + Fagaalu.Control_28 + Fagatele.Control_28)/3, levels=design)

```


```{r heatvcontrol}
#Look at genes expressed  bw heat and control 
test.anov.one <- glmQLFTest(fit, contrast= con.heatvscontrol)
summary(decideTests(test.anov.one))
#Write plot to file
jpeg("glmQLF_heatvscontrol_plotMD.jpg")
plotMD(test.anov.one)
abline(h=c(-1, 1), col="blue")
dev.off()
#Write top tags table of DE genes to file
tagsTblANOVA.one <- topTags(test.anov.one, n=nrow(test.anov.one$table))$table
tagsTblANOVA.one.keep <- tagsTblANOVA.one$FDR <= 0.05
tagsTblANOVA.one.out <- tagsTblANOVA.one[tagsTblANOVA.one.keep,]
write.table(tagsTblANOVA.one.out, file="glmQLF_UVvsVIS_topTags.csv", sep=",", row.names=TRUE)
```

Same as above but filtered this time i think

```{r filtered_tags}
#Look at genes with significant expression across all groups
treat.anov.one <- glmTreat(fit, contrast=con.heatvscontrol, lfc=log2(1.2))
summary(decideTests(treat.anov.one))
#Write plot to file
jpeg("glmQLF_UVvsVIS_plotMD_filtered.jpg")
plotMD(treat.anov.one)
abline(h=c(-1, 1), col="blue")
dev.off()
#Write tags table of DE genes to file
tagsTblANOVA.one.filtered <- topTags(treat.anov.one, n=nrow(treat.anov.one$table))$table
tagsTblANOVA.one.filtered.keep <- tagsTblANOVA.one.filtered$FDR <= 0.05
tagsTblANOVA.one.filtered.out <- tagsTblANOVA.one.filtered[tagsTblANOVA.one.filtered.keep,]
write.table(tagsTblANOVA.one.filtered.out, file="glmQLF_UVvsVIS_topTags_filtered.csv", sep=",", row.names=TRUE)
```


Question: how to do this for 3 groups? in the examples it's always contrasts between 2 groups. But I have 3 sites!! 
-could make 3 contrasts 


Question: maybe try paired design and pair by genotype?


